src/exporters/coreml/__main__.py

```diff
@@ -121,14 +123,21 @@ def main():
         preprocessor = AutoProcessor.from_pretrained(args.model)
     else:
         raise ValueError(f"Unknown preprocessor type '{args.preprocessor}'")
+    
+    # Support legacy task names in CLI only
+    feature = args.feature
+    args.feature = FeaturesManager.map_from_synonym(args.feature)
+    if feature != args.feature:
+        deprecation_message = f"Feature '{feature}' is deprecated, please use '{args.feature}' instead."
+        warnings.warn(deprecation_message, FutureWarning)
 
     # Allocate the model
     model = FeaturesManager.get_model_from_feature(
         args.feature, args.model, framework=args.framework, #cache_dir=args.cache_dir
     )
     model_kind, model_coreml_config = FeaturesManager.check_supported_model_or_raise(model, feature=args.feature)
 
-    if args.feature in ["seq2seq-lm", "speech-seq2seq"]:
```

Comment:
For `speech-seq2seq`, I renamed it as well to `automatic-speech-recognition` to align with https://huggingface.co/datasets/huggingface/transformers-metadata/blob/main/pipeline_tags.json

Reply:
Yes, I saw that but found it a bit confusing to deal with that single special case here. I'd rather keep them separate for simplicity and we'll address the change when we move to Optimum, if that's ok.

I did change some inconsistent use of both `"speech-recognition"` and `"automatic-speech-recognition"` in the previous commit.